### PR TITLE
Repair address completion

### LINKED ISSUE #1213: PERSON: Adressvervollständigung testen
Tolle Funktion - Wir freuen uns! Zwei Dinge fallen mir auf:

### Nicht gefundene Adressen
Einige Adressen sind nicht bekannt, bzw. erscheinen nicht im Autocomplete. Beispiele: 

- Seftigenstrasse 10b, 3007 Bern
- Stoossstrasse 2, 3008 Bern

image
'
## Analyse-Task
- [x] Setup Doku Adressvervollständigung erstellen bzw. ergänzen falls bereits vorhanden
- [x] Prüfen, warum die oben genannten Adressen nicht korrekt importiert werden.
- [x] In den Übersetzungen von den Tags "Addresse befindet sich nicht im Verzeichnis" --> "Adresse befindet sich nicht im Verzeichnis"
- [x] Prüfen, warum bei der PBS-Integrationsumgebung die "falschen" Adressen nicht getagt werden.
  - https://pbs.puzzle.ch/de/groups/1/people/777.html
    - fehlerhafte Adresse, doch kommt kein Tag 
    - Ev. auch mit dem Buchstabenproblem gelöst
- [x] Entweder fixen, wenn kleine Sache, oder deaktivieren für Release PBS

### PR DESCRIPTION
**Pull request by kronn**
While debugging the address completion, I refactored the codebase a little. Therefore, you are now looking at an extraction, some spelling changes, several linter-changes and some new methods. Alongside these cosmetic changes, I added

- support for house numbers as they are imported right now
- some documentation on this optional feature
- a spec that ensures consistent spellings in the German locale

Fixes #1213

### CODE REVIEW

File: spec/config/locales_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    # the spelling "email" has a few false positives :-)
+    email_lines = `grep -e ':.*email' config/locales/*.de.yml`
+                  .split("\n").map(&:chomp)
+                  .reject { |line| line =~ /.*Scope: email.*/ }
+                  .reject { |line| line =~ /.*%{email}.*/ }
+    expect(email_lines).to be_empty
+  end
+
+  it 'do not contain wrong spellings of Adress' do
```
Comment (psunix) [inline spec/config/locales_spec.rb:24]:
```suggestion
  it 'do not contain wrong spellings of address' do
```

File: app/domain/address/full_text_search.rb [hunk 2]
```diff
@@ -1,15 +1,15 @@
 # frozen_string_literal: true
 
-#  Copyright (c) 2012-2020, CVP Schweiz. This file is part of
-#  hitobito_cvp and licensed under the Affero General Public License version 3
+#  Copyright (c) 2012-2021, CVP Schweiz. This file is part of
```
Comment (psunix) [inline app/domain/address/full_text_search.rb:3]: es war einmal CVP ... jetzt mittig ...

File: doc/features/address-completion.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Addresses
+
+[TOC]
+
+## Overview
+
+It is possible to provide an auto-complete and a validation for addresses.
+The Addresses are read from the Address model.
+
+Currently, address entries are swiss addresses imported from the Swiss Post.
```
Comment (psunix) [inline doc/features/address-completion.md:10]:
```suggestion
Currently, address entries are swiss addresses imported from Swiss Post.
```

File: doc/features/address-completion.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+- imported_countries (a list of countries to be imported, currently only 'CH')
+
+## Completion
+
+In the person#edit view, the address is auto-completed from all entries in the
+addresses-table. If Sphinx is present, sphinx indexes that table and answers
+queries. Otherwise, the table is queried by the database.
+
+The search-strategy is determined the same way as for the
+people/group/event-full-text search.
```
Comment (psunix) [inline doc/features/address-completion.md:37]: maybe link to file?

File: doc/features/address-completion.md (hunk 3 above)
Comment (psunix) [reply doc/features/address-completion.md:10]: maybe leave a comment / link on where to get this token from Swiss Post? is it free?

**Review by psunix — APPROVED** [review]
:+1: :vulcan_salute: :partying_face:

File: doc/features/address-completion.md (hunk 3 above)
Comment (kronn) [reply doc/features/address-completion.md:10]: That is something I do not know and could not find any information about. I want to ask @TheWalkingLeek once I get a hold of him.

File: app/domain/address/full_text_search.rb (hunk 2 above)
Comment (kronn) [reply app/domain/address/full_text_search.rb:3]: davon gibt es noch einige. Ich mache dazu einen eigenen PR.

**Pull request merged**